spek-extension/robolectric/src/test/kotlin/robolectric/InstanceFactoryTest.kt

```diff
@@ -0,0 +1,95 @@
+package robolectric
+
+import android.app.Activity
+import android.app.IntentService
+import android.content.Intent
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.robolectric.Robolectric
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+import org.robolectric.internal.SdkEnvironment
+import org.robolectric.manifest.AndroidManifest
+import org.spekframework.spek2.CreateWith
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.lifecycle.InstanceFactory
+import org.spekframework.spek2.robolectric.RobolectricInstanceFactory
+import org.spekframework.spek2.robolectric.RobolectricLifeCycleListener
+import org.spekframework.spek2.runtime.test.AbstractSpekRuntimeTest
+
+class RobolectricInstanceFactoryTest : AbstractSpekRuntimeTest() {
+
+    object SampleInstanceFactoryTest : InstanceFactory by RobolectricInstanceFactory
+
+    @Test
+    fun testUsingRobolectric() {
+        @CreateWith(SampleInstanceFactoryTest::class)
+        @Config(sdk = intArrayOf(21))
+        class RobolectricSpek(val sdkEnvironment: SdkEnvironment, val config: Config, val androidManifest: AndroidManifest ): Spek({
+            lateinit var activity: TestActivity1
+            registerListener(RobolectricLifeCycleListener(sdkEnvironment, config, androidManifest))
```

Comment:
I guess we could provide `abstract class RobolectriSpek` to remove this from user's code